plugins/chat/assets/javascripts/discourse/components/chat/thread/header.js

```diff
@@ -22,9 +26,49 @@ export default class ChatThreadHeader extends Component {
     );
   }
 
+  get threadNotificationLevel() {
+    return this.membership?.notificationLevel || NotificationLevels.REGULAR;
+  }
+
+  get membership() {
+    return this.args.thread.currentUserMembership;
+  }
+
   @action
   openThreadSettings() {
     const controller = showModal("chat-thread-settings-modal");
     controller.set("thread", this.args.thread);
   }
+
+  @action
+  updateThreadNotificationLevel(val) {
+    let originalVal;
+
+    if (this.membership) {
+      originalVal = this.membership.notificationLevel;
+      this.membership.notificationLevel = val;
+    } else {
+      this.args.thread.currentUserMembership = UserChatThreadMembership.create({
+        notification_level: val,
+        last_read_message_id: null,
+      });
+    }
+
+    this.chatApi
+      .updateCurrentUserThreadNotificationsSettings(
+        this.args.thread.channel.id,
+        this.args.thread.id,
+        { notificationLevel: val }
+      )
+      .then((response) => {
+        this.membership.last_read_message_id =
+          response.membership.last_read_message_id;
+      })
+      .catch((err) => {
+        if (this.membership && originalVal) {
```

Comment:
I don't think there's a chance we don't have `this.membership` here

Reply:
Oh yes good point +1